robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowImageView.java

```diff
@@ -1,72 +1,52 @@
 package org.robolectric.shadows;
 
 import android.graphics.Bitmap;
-import android.graphics.Canvas;
 import android.graphics.Matrix;
-import android.graphics.drawable.BitmapDrawable;
-import android.graphics.drawable.Drawable;
 import android.widget.ImageView;
 import org.robolectric.Shadows;
 import org.robolectric.annotation.Implementation;
 import org.robolectric.annotation.Implements;
+import org.robolectric.annotation.RealObject;
+
+import static org.robolectric.internal.Shadow.directlyOn;
 
 /**
  * Shadow for {@link android.widget.ImageView}.
  */
 @Implements(ImageView.class)
 public class ShadowImageView extends ShadowView {
-  private Drawable imageDrawable;
   private Bitmap imageBitmap;
   private ImageView.ScaleType scaleType;
   private Matrix matrix = new Matrix();
   private int imageLevel;
 
+  @RealObject
+  protected ImageView realImageView;
+
+  public ShadowImageView() {
+  }
+
   @Implementation
   public void setImageBitmap(Bitmap imageBitmap) {
-    setImageDrawable(new BitmapDrawable(imageBitmap));
     this.imageBitmap = imageBitmap;
+
+    directlyOn(realImageView, ImageView.class).setImageBitmap(imageBitmap);
   }
 
+  /**
+   * @deprecated
+   */
   @Deprecated
   public Bitmap getImageBitmap() {
     return imageBitmap;
   }
 
-  @Implementation
-  public void setImageDrawable(Drawable drawable) {
-
-    if (this.imageDrawable != drawable) {
-      this.imageDrawable = drawable;
-      invalidate();
-    }
-  }
-
-  @Implementation
-  public void setImageResource(int resId) {
-    setImageDrawable(resId != 0 ? buildDrawable(resId) : null);
-  }
-
   public int getImageResourceId() {
-    ShadowDrawable shadow = Shadows.shadowOf(imageDrawable);
+    ShadowDrawable shadow = Shadows.shadowOf(realImageView.getDrawable());
     return shadow.getCreatedFromResId();
   }
 
   @Implementation
-  public ImageView.ScaleType getScaleType() {
-    return scaleType;
-  }
-
-  @Implementation
-  public void setScaleType(ImageView.ScaleType scaleType) {
-    this.scaleType = scaleType;
-  }
-
-  @Implementation
-  public Drawable getDrawable() {
-    return imageDrawable;
-  }
-
-  @Implementation
   public void setImageMatrix(Matrix matrix) {
```

Comment:
Why not just call through here? Or remove the method entirely. I think it should just work if you remove both accessor and mutator?